File: OCaml/LED_Numbers.ml
open Core.Std

let get_leds number =
let leds = [0, [" _  ";"| | ";"|_| "];
    	    1, ["  ";"| ";"| "];
			2, [" _  ";" _| ";"|_  "];
			3, ["_  ";"_| ";"_| "];
            4, ["    ";"|_| ";"  | "];
            5, [" _  ";"|_  ";" _| "];
            6, [" _  ";"|_  ";"|_| "];
			7, ["_   ";" |  ";" |  "];
            8, [" _  ";"|_| ";"|_| "];
            9, [" _  ";"|_| ";" _| "]] in
	for i = 0 to 2 do
		for j = 0 to String.length(number) - 1 do
			let line = List.Assoc.find_exn leds 
					   (int_of_string(Char.to_string(number.[j]))) in
			printf "%s" (List.nth_exn line i)
		done;
		print_string "\n"
	done
	
let () =
	print_string "Enter a number: "
	let num = read_line() in
	get_leds num

File: Scala/LED_Numbers.scala
@main def ledNumbers(num: Int) =
for( x <- 0 to 2 ) {
    for( y <- 0 to num.toString.length - 1 ) {
	val s_num = num.toString
	print(leds(s_num(y).toString)(x))
    }
    println("")
}

val leds = Map("0" -> List(" _  ","| | ","|_| "),
               "1" -> List("  ","| ","| "),
               "2" -> List(" _  "," _| ","|_  "),
               "3" -> List("_  ","_| ","_| "),
               "4" -> List("    ","|_| ","  | "),
               "5" -> List(" _  ","|_  "," _| "),
               "6" -> List(" _  ","|_  ","|_| "),
               "7" -> List("_   "," |  "," |  "),
               "8" -> List(" _  ","|_| ","|_| "),
               "9" -> List(" _  ","|_| "," _| "))
